### PR TITLE
feat(query): CNI Plugin Does Not Support Network Policies for Kubernetes

### PR DESCRIPTION
**Pull request by cx-miguel-dasilva**
**Proposed Changes**
- CNI Plugin Does Not Support Network Policies for Kubernetes
- Add support to CNI files in the analyzer 

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

**Review by rjegoncalves — COMMENTED** [review]
The severity and category LGTM.
The description should be better aligned with the CIS description (_5.3.1 Ensure that the CNI in use supports Network Policies_).

File: assets/queries/k8s/cni_plugin_does_not_support_network_policies/query.rego [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	plugin := document.plugins[j]
+	plugin.type == "flannel"
+
+	result := {
+		"documentId": document.id,
+		"searchKey": sprintf("plugins", []),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("Plugins should not contain a plugin that does not support Network Policies", []),
+		"keyActualValue": sprintf("Plugins contains a plugin that does not support Network Policies", []),
+	}
```
Comment (rafaela-soares) [inline assets/queries/k8s/cni_plugin_does_not_support_network_policies/query.rego:19]: Please, add `searchLine`:

`"searchLine": common_lib.build_search_line(["plugins", j, "type"], []),`

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

File: assets/queries/k8s/cni_plugin_does_not_support_network_policies/query.rego [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	common_lib.valid_key(document, "cniVersion")
+	plugin := document.plugins[j]
+	plugin.type == "flannel"
+
+	result := {
+		"documentId": document.id,
+		"searchKey": sprintf("plugins", []),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("Plugins should not contain a plugin that does not support Network Policies", []),
+		"keyActualValue": sprintf("Plugins contains a plugin that does not support Network Policies", []),
```
Comment (joaorufi) [inline assets/queries/k8s/cni_plugin_does_not_support_network_policies/query.rego:18]:
```suggestion
		"keyExpectedValue": "Plugins should not contain a plugin that does not support Network Policies",
		"keyActualValue": "Plugins contains a plugin that does not support Network Policies",
```

File: assets/queries/k8s/cni_plugin_does_not_support_network_policies/query.rego [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	cni:= json.unmarshal(document.data["cni-conf.json"])
+    plugin := cni.plugins[j]
+    plugin.type == "flannel"
+
+	result := {
+		"documentId": document.id,
+		"searchKey": sprintf("data.cni-conf.json", []),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("Plugins should not contain a plugin that does not support Network Policies", []),
+		"keyActualValue": sprintf("Plugins contains a plugin that does not support Network Policies", []),
```
Comment (joaorufi) [inline assets/queries/k8s/cni_plugin_does_not_support_network_policies/query.rego:35]:
```suggestion
		"keyExpectedValue": "Plugins should not contain a plugin that does not support Network Policies",
		"keyActualValue": "Plugins contains a plugin that does not support Network Policies",
```

**Review by joaorufi — CHANGES_REQUESTED** [review]

**Review by joaorufi — APPROVED** [review]

**Pull request merged**